.github/workflows/build-and-test.yml

```diff
@@ -21,12 +21,14 @@ jobs:
       - name: Installing dependencies and building packages
         uses: ./.github/actions/ci
 
+      - name: Lint
+        run: pnpm lint
+
+      - name: Typecheck
+        run: pnpm typecheck:strict
+
```

Comment:
As of this PR, we don't have lint or type errors, we might as well keep it that way with a CI check. We also might as well run these first, so that we don't spend compute time running tests if the workflow is going to fail.